plugin/code_actions.py

```diff
@@ -108,11 +108,18 @@ def show_popup_menu(self) -> None:
 
     def handle_select(self, index: int) -> None:
         if index > -1:
-            client = client_for_view(self.view)
-            if client:
-                client.send_request(
-                    Request.executeCommand(self.commands[index]),
-                    self.handle_command_response)
-
-    def handle_command_response(self, response):
-        pass
+            # TODO for CodeAction, handle potential WorkspaceEdit before Command.
+            result = self.commands[index]
+            command_attr = result.get('command', None)
+            # Result can be either a Command or CodeAction type.
+            command = result if isinstance(command_attr, str) else command_attr
+            if command:
+                command_name = command.get('command')
+                command_args = command.get('arguments', None)
+
+                args = {
+                    "command_name": command_name,
+                    "command_args": command_args,
+                    "modal_result": False,
+                }
+                self.view.run_command('lsp_execute', args)
```

Comment:
Would prefer if the command execution didn't use `LspExecuteCommand`, that class is for letting the user directly run commands from the quick panel and not to be invoked from code.

It's fine if we want them to share functionality though!